### PR TITLE
Misc compile include/warn fixes

### PR DESCRIPTION
**Pull request by travisdowns**
Fix wrong includes related to SEASTAR_MODULE
Fix "unused result" warnings in gcc

Tests all pass, see changes for details.

### CODE REVIEW

File: tests/perf/linux_perf_event.cc [hunk 1]
```diff
@@ -61,7 +63,8 @@ linux_perf_event::read() {
         return 0;
     }
     uint64_t ret;
-    ::read(_fd, &ret, sizeof(ret));
+    auto res =::read(_fd, &ret, sizeof(ret));
```
Comment (tchaikov) [inline tests/perf/linux_perf_event.cc:66]: nit, add a space after `=`.

File: src/core/reactor.cc [hunk 2]
```diff
@@ -2916,7 +2917,8 @@ void reactor::start_aio_eventfd_loop() {
     future<> loop_done = repeat([this] {
         return _aio_eventfd->readable().then([this] {
             char garbage[8];
-            ::read(_aio_eventfd->get_fd(), garbage, 8); // totally uninteresting
+            auto res = ::read(_aio_eventfd->get_fd(), garbage, 8); // totally uninteresting
+            (void)res;
```
Comment (tchaikov) [inline src/core/reactor.cc:2921]: nit, how about using `std::ignore`? i imagine if we can replace all of them with `_` in C++26. as `std::ignore` is easier to grep than `void` , and visually more obvious than the later. so i prefer `std::ignore`.

**Review by tchaikov — APPROVED** [review]
thanks. just two nits.

for posterity, we pass `-UNDEBUG` when building seastar, so cannot use `assert()` on the critical pathes.

**Comment by travisdowns** [comment]
> for posterity, we pass `-UNDEBUG` when building seastar, so cannot use `assert()` on the critical pathes.

Yes, thanks for the note. I don't think any of the changes here involve a location that would be a performance concern for `assert`.

File: src/core/reactor.cc (hunk 2 above)
Comment (travisdowns) [reply src/core/reactor.cc]: Thanks of the note, changed to `std::ignore`.

File: tests/perf/linux_perf_event.cc (hunk 1 above)
Comment (travisdowns) [reply tests/perf/linux_perf_event.cc]: Fixed, thanks!

**Comment by tchaikov** [comment]
> > for posterity, we pass `-UNDEBUG` when building seastar, so cannot use `assert()` on the critical pathes.
> 
> Yes, thanks for the note. I don't think any of the changes here involve a location that would be a performance concern for `assert`.

ahh, thought that the `start_aio_eventfd_loop()` one was the loop for reaping the events, i was wrong.

**Review by tchaikov — APPROVED** [review]
lgtm

**Comment by tchaikov** [comment]
@scylladb/seastar-maint hello maintainers, could you help review/merge this change?

**Comment by tchaikov** [comment]
@scylladb/seastar-maint hello maintainers, could you help review/merge this change?

**Pull request closed without merging**